src/transformers/testing_utils.py

```diff
@@ -3094,6 +3096,13 @@ def get_device_properties() -> DeviceProperties:
             return ("rocm", major)
         else:
             return ("cuda", major)
+    elif IS_XPU_SYSTEM:
+        import torch
+
+        arch = torch.xpu.get_device_capability()["architecture"]
+        gen_mask = 0x000000FF00000000
+        gen = (arch & gen_mask) >> 32
```

Comment:
maybe just a comment here to explain what this does  and/or a link to a documentation page if any?

Reply:
sure, done